lib/utils/compose.coffee

```diff
@@ -0,0 +1,732 @@
+				.then (stream) ->
+					task.buildStream = stream
+				.return([ task, binPath ])
+	.map ([ task, qemuPath ]) ->
+		Promise.resolve(task).tap (task) ->
+			captureStream = buildLogCapture(task.external, task.logBuffer)
+
+			if task.external
+				# External image -- there's no build to be performed,
+				# just follow pull progress.
+				captureStream.pipe(task.logStream)
+				task.progressHook = pullProgressAdapter(captureStream)
+			else
+				task.streamHook = (stream) ->
+					if qemuPath?
+						buildThroughStream = transpose.getBuildThroughStream
+							hostQemuPath: qemuPath
+							containerQemuPath: "/tmp/#{qemu.QEMU_BIN_NAME}"
+						rawStream = stream.pipe(buildThroughStream)
+					else
+						rawStream = stream
+					# `stream` sends out raw strings in contrast to `task.progressHook`
+					# where we're given objects. capture these strgins as they come
+					# before we parse them.
+					rawStream
+						.pipe(captureStream)
+						.pipe(buildProgressAdapter(inlineLogs))
+						.pipe(task.logStream)
+	.then (tasks) ->
+		logger.logDebug 'Prepared tasks; building...'
+		builder.performBuilds(tasks, docker)
+		.map (builtImage) ->
+			if not builtImage.successful
+				builtImage.error.serviceName = builtImage.serviceName
+				throw builtImage.error
+
+			d = imageDescriptorsByServiceName[builtImage.serviceName]
+			task = _.find(tasks, serviceName: builtImage.serviceName)
+
+			image =
+				serviceName: d.serviceName
+				name: d.image.tag ? d.image
+				logs: task.logBuffer.join('\n')
+				props:
+					dockerfile: builtImage.dockerfile
+					projectType: builtImage.projectType
+
+			# Times here are timestamps, so test whether they're null
+			# before creating a date out of them, as `new Date(null)`
+			# creates a date representing UNIX time 0.
+			if (startTime = builtImage.startTime)
+				image.props.startTime = new Date(startTime)
+			if (endTime = builtImage.endTime)
+				image.props.endTime = new Date(endTime)
+			docker.getImage(image.name).inspect().get('Size').then (size) ->
+				image.props.size = size
+			.return(image)
+		.tap (images) ->
+			summary = _(images).map ({ serviceName, props }) ->
+				[ serviceName, "Image size: #{humanize.filesize(props.size)}" ]
+			.fromPairs()
+			.value()
+			renderer.end(summary)
+	.finally(renderer.end)
+
+createRelease = (apiEndpoint, auth, userId, appId, composition) ->
+	_ = require('lodash')
+	crypto = require('crypto')
+	releaseMod = require('resin-release')
+	versions = require('./versions')
+
+	client = releaseMod.createClient({ apiEndpoint, auth })
+
+	releaseMod.create
+		client: client
+		user: userId
+		application: appId
+		composition: composition
+		source: "cli/#{versions.cli.major}"
+		commit: crypto.pseudoRandomBytes(16).toString('hex').toLowerCase()
+	.then ({ release, serviceImages }) ->
+		[ 'created_at', 'belongs_to__application', 'is_created_by__user', '__metadata' ].forEach (attr) ->
+			delete release[attr]
```

Comment:
Would it make sense to `pick` only the properties that we are interested in instead of removing those that we are not?

Reply:
The issue here is that these properties (being relations to resources, not "primitive" properties) are problematic when sent to the API (it fails in an awful way that leaves you scratching your head) so I'd rather make this clear.